### PR TITLE
New Spezi Standard Implementation

### PR DESCRIPTION
**Pull request by niallkehoe**
# New Spezi Standard Implementation

## :recycle: Current situation & Problem
Currently, the SpeziHealthKit module lacks support for the new Spezi Standard released in Spezi 0.7.0
*This PR implements the improvements mentioned in [https://github.com/https://github.com/StanfordSpezi/Spezi/pull/73](Pull 73) *

## :bulb: Proposed solution
Update the codebase to the newest Spezi 0.7.0

## :gear: Release Notes 
This feature will allow for greater flexibility when using Spezi Questionnaire.
```
/// A Constraint which all `Standard` instances must conform to when using the Spezi HealthKit module.
public protocol QuestionnaireConstraint: Standard {
    /// Adds a new `HKSample` to the ``HealthKit`` module
    /// - Parameter response: The `HKSample` that should be added.
    func add(_ response: HKSample) async
    
    /// Removes a `HKSampleRemovalContext` from the ``HealthKit`` module
    /// - Parameter response: The `HKSampleRemovalContext` that contains information on the item that should be removed.
    func remove(removalContext: HKSampleRemovalContext)
}
```

### Related PRs
See [https://github.com/https://github.com/StanfordSpezi/Spezi/pull/73](Pull 73) in Spezi for details

### Testing
Passes all tests

### Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md):
- [✅ ] I agree to follow the [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: Tests/UITests/TestApp/TestApp.swift
Comment (PSchmiedmayer) [inline Tests/UITests/TestApp/TestApp.swift:1]: Your UI tests are currently failing as the `hkTypeIdentifierAssert` methods and others expect a list of HealthKit sample types compared to the UUIDs you are currently displaying in the application:
Screenshot 2023-08-06 at 11 51 08 AM

You will need to change the logic in the UI in a way that it takes a look at the stored `HKSample` instances (see other comments) and then displays the sample type in the UI for the UI tests to pick this up and count them. I would recommend taking a look at the previous implementation that extracted the sample type as Strings as well as the test methods.

File: Tests/UITests/TestApp/HealthKitTestsView.swift [hunk 1]
```diff
@@ -13,8 +13,9 @@ import XCTSpezi
 
 
 struct HealthKitTestsView: View {
-    @EnvironmentObject var healthKitComponent: HealthKit<TestAppStandard>
-    @EnvironmentObject var standard: TestAppStandard
+    @EnvironmentObject var healthKitComponent: HealthKit
+    @EnvironmentObject var standard: ExampleStandard
+    
     @State var dataChanges: [String] = []
```
Comment (PSchmiedmayer) [inline Tests/UITests/TestApp/HealthKitTestsView.swift:19]: Referencing my comment on `Tests/UITests/TestApp/TestApp.swift`, I would suggest that you extract the sample type (https://developer.apple.com/documentation/healthkit/hksample/1615744-sampletype) here and display that in the UI (e.g. using the `identifier` property: https://developer.apple.com/documentation/healthkit/hkobjecttype/1615294-identifier). 

This should show the element type in the UI and allows the UI tests to pick this up 👍

**Review by PSchmiedmayer — CHANGES_REQUESTED** [review]
@niallkehoe Thank you for the PR and the improvements to the HealthKit module! This really demonstrates how the new `Standard` implementation improves these types of interactions and how this leads to a cleaner UI for the `Standard`s. 

I had several comments across the PR, let me know if you have any questions or comments regarding the different elements! Happy to go into detail if you have any follow-up questions.

**Comment by PSchmiedmayer** [comment]
@niallkehoe Thank you for continuing to work on the PR, let me know once you have addressed all comments and you think that it is in a good shape to give it an other review 🚀

**Comment by niallkehoe** [comment]
@Paul Just uploaded my code after a check of all the files

**Review by PSchmiedmayer — APPROVED** [review]
Thank you for the improvements @niallkehoe! I only had a few very small comments that would be great to be addressed before we merge the PR. No need for an other review, you can merge the PR one the conversations are resolved.

Thank you for all the work and effort that went into improving Spezi and the HealthKit module!

**Pull request merged**